### PR TITLE
Revise release-a-new-version.md to the Gitflow workflow

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Revised release-a-new-version.md to the Gitflow workflow.

Refer to: https://stellarfoundation.slack.com/archives/C02TPN2H65N/p1686605671225359

### CODE REVIEW

**Review by philipliu — APPROVED** [review]
What about releasing on a holiday?
`- [ ] DO NOT RELEASE before holidays or weekends! Mondays and Tuesdays are preferred.`

We plan to release 2.1.0 on Monday, but it's Juneteenth in the US.

File: .github/ISSUE_TEMPLATE/release-a-new-version.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ assignees: ''
 - [ ] Check the docker image of the release automatically published to [Docker Hub](https://hub.docker.com/r/stellar/anchor-platform).
 - [ ] If necessary, update the badges versions in [docs/00 - Stellar Anchor Platform.md].
 - [ ] Create the pull request `release/2.0.x -> main`: this should require two approvals. DO NOT squash merge.   
-- [ ] Create another pull request `main -> develop`: AFTER the release branch is merged with the `main` branch. DO NOT squash merge.
+- [ ] Create another pull request `release/2.0.x -> develop`: AFTER the release branch is merged with the `main` branch. DO NOT squash merge.
```
Comment (Ifropc) [inline .github/ISSUE_TEMPLATE/release-a-new-version.md:28]: Won't it have the same issue we had before, where main has a merge commit that develop doesn't have?

Comment (lijamie98) [reply .github/ISSUE_TEMPLATE/release-a-new-version.md:28]: We squashed merged to `main` in the previous release which caused the conflict.

Comment (Ifropc) [reply .github/ISSUE_TEMPLATE/release-a-new-version.md:28]: So we don't squash merge to main anymore right?

Comment (lijamie98) [reply .github/ISSUE_TEMPLATE/release-a-new-version.md:28]: I think squash merge to `main` is troublesome.

**Review by Ifropc — APPROVED** [review]

**Pull request merged**